### DISCUSSION TITLE
Apache pulsar concurrency for Key_Shared consumer

### DISCUSSION THREAD

**Discussion by robp94** [smallrye/smallrye-reactive-messaging#2718] [Q&A] [answered]
If I have a Key_Shared consumer, are message with different keys run in parallel?

**Comment by ozangunalp** [comment, marked as answer]
No, but to achieve this, you can check the [concurrency](https://smallrye.io/smallrye-reactive-messaging/latest/concepts/incoming-concurrency/) attribute, which creates multiple copies of the incoming channel.
